app_dart/bin/server.dart

```diff
@@ -82,29 +93,135 @@ Future<void> main() async {
         authProvider,
         scheduler,
       ),
+
+      /// Updates cocoon task status.
+      ///
+      /// This API updates task status in datastore.
+      ///
+      /// POST: /api-update-status
+      ///
+      /// Parameters:
+      ///   CommitBranch: (string in body). Branch of commit.
+      ///   CommitSha: (string in body). Sha of commit.
+      ///   BuilderName: (string in body). Name of the luci builder.
+      ///   NewStatus: (string in body) required. Status of the task.
+      ///   ResultData: (string in body) optional. Benchmark data.
+      ///   BenchmarkScoreKeys: (string in body) optional. Benchmark data.
+      ///
+      /// Response: Status 200 OK
+      ///
+      /// Requires authentication: Status: 401 Unauthorized
```

Comment:
We should separate what requires authentication from the response as well (for example, update-task-status only allows the swarming bots to update tasks)

Reply:
Note sure what you mean here? Are you saying to add a note that the update-task-status is only authenticated by the bots? Can you show an example comment you want here?

Reply:
These are intended more as api user docs rather than implementation, I'm following what seems to be general api doc principles (https://docs.github.com/en/rest/reference/actions) where potential responses are stated for user.

Maybe I should add this under a "Responses" heading to make it more clear?

Reply:
Since this is a Flutter owned project, the Flutter and Dart style guides will take precedence. An example of how we can balance the two styles:

```
/// Returns [HttpStatus.ok] if a corresponding task exists, can be updated, and metrics center can push performance payload to skia-perf.
/// If [authProvider] cannot authenticate the current request, [HttpStatus.forbidden] will be returned.
```

As a side note, I'm hesitant to include auth info as it's dependent on the auth provider (some return both 401's and 403's)